Dockerfile

```diff
@@ -14,12 +14,15 @@ RUN rm src/lib.rs
 COPY src/* src/
 # build the executable
 RUN cargo build --release
+# download jq
+RUN curl -L -o jq https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64 \
```

Comment:
Not critical, but right now, this isn't optimally using the docker cache: we expect the test runner to change more often than jq. It's not used in this stage anyway, so there's no real reason to download it here. I'd put this between the FROM statement in the test stage and the COPY statements.

Reply:
Right! I actually put it there to avoid rebuild and forget to put it back later XD

Reply:
Oh Actually I misunderstood your comment ... I thought I should download in the build stage to avoid an additional layer in the test stage.

Reply:
I can see 3 better locations than currently.
1. just after `WORKDIR` in build stage (and keep copy in test stage)
2. just after `WORKDIR` in test stage (and remove copy)
3. appended to the existing `RUN` in test stage to avoid additional layer (and remove copy)

Which one do you prefer?

Reply:
I like option 2. I'm usually much more concerned about avoiding spurious cache misses than about minimizing the total number of layers.